core/src/core/transaction.rs

```diff
@@ -469,15 +537,21 @@ impl Summable for SumCommit {
 
 impl Writeable for SumCommit {
 	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ser::Error> {
-		self.commit.write(writer)
+		self.commit.write(writer)?;
+		self.switch_commit_hash.write(writer)?;
+		Ok(())
 	}
 }
 
 impl Readable for SumCommit {
 	fn read(reader: &mut Reader) -> Result<SumCommit, ser::Error> {
 		let secp = secp::Secp256k1::with_caps(secp::ContextFlag::Commit);
+		let commit = Commitment::read(reader)?;
+		let switch_commit_hash = SwitchCommitHash::read(reader)?;
+
 		Ok(SumCommit {
-			commit: Commitment::read(reader)?,
+			commit: commit,
+			switch_commit_hash: switch_commit_hash,
 			secp: secp,
```

Comment:
Just reading through this - not directly related to your changes, but do we need to maintain an instance of `secp` in every `SumCommit`?
Seems like this the only place we actually use this is in `add` in `impl ops::Add for SumCommit` (and we could just init `secp` there on demand?)